src/main/java/com/elastic/support/diagnostics/commands/SystemCallsCmd.java

```diff
@@ -1,25 +1,232 @@
+            return;
+        } else if (context.isDocker()) {
+            // If docker is running and we got an id passed in check to see if it's
+            // in the list of container id's we obtained in the host check. If it is
+            // just get that one. If it's not there or there was no id go get all the containers in the list.
+            List<String> containers = context.getDockerContainers();
+            String dockerId = context.getDiagnosticInputs().getDockerId();
+            if (StringUtils.isNotEmpty(dockerId)) {
+                boolean isDockerIdInList = containers.contains(dockerId);
+                if (isDockerIdInList) {
+                    containers.clear();
+                    containers.add(dockerId);
+                }
+            }
+
+            // Run the non-container specific docker calls
+            Map<String, String> dockerGlobal = context.getDiagsConfig().getCommandMap("docker-global");
+            processCalls(tempDir, dockerGlobal, pb);
+
+            Map<String, String> dockerContainers = context.getDiagsConfig().getCommandMap("docker-containers");
+            // Now get the stats for each container in the list. If they passed one in and we found it this will
+            // be a single iteration
+            for (String container : containers) {
+                try {
+                    String dir = context.getTempDir() + SystemProperties.fileSeparator + container;
+                    Files.createDirectories(Paths.get(dir));
+                    Map<String, String> containerCalls = preProcessDockerCommands(dockerContainers, container);
+                    processCalls(dir, containerCalls, pb);
+                } catch (IOException e) {
+                    logger.log(SystemProperties.DIAG, e);
+                    logger.info("Error creating container directory - bypassing Docker container calls.");
+                }
+            }
+        } else {
+            String calls = checkOS();
+            Map<String, String> osCmds = context.getDiagsConfig().getCommandMap(calls);
+            osCmds = preProcessOsCmds(osCmds, pid, SystemProperties.javaHome);
+            processCalls(tempDir, osCmds, pb);
+        }
+
+    }
+
+    public void processCalls(String targetDir, Map<String, String> commandMap, ProcessBuilder pb) {
+
+        commandMap.forEach((k, v) -> {
+                    runCommand(targetDir, k, v, pb);
+                }
+        );
+    }
+
+    public String checkOS() {
+        String osName = SystemProperties.osName.toLowerCase();
+        if (osName.contains("windows")) {
+            return "winOS";
+        } else if (osName.contains("linux")) {
+            return "linuxOS";
+        } else if (osName.contains("darwin") || osName.contains("mac os x")) {
+            return "macOS";
+        } else {
+            logger.error("Failed to detect operating system!");
```

Comment:
```suggestion
            logger.error("Error: unrecognized operating system [{}]", osName);
```

This way we get it here if it's sent back without paying attention.

Reply:
Addressing this in the next comment. I still want this to error out but I'm going to handle it a little more leniently.